src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala

```diff
@@ -168,7 +167,11 @@ trait LibraryServiceSupport extends DataUseRestrictionSupport with LazyLogging {
   }
 
   def getEffectiveDiscoverGroups(samDAO: SamDAO)(implicit ec: ExecutionContext, userInfo:UserInfo): Future[Seq[String]] = {
-    samDAO.listGroups(userInfo) map {FireCloudConfig.ElasticSearch.discoverGroupNames intersect _}
+    samDAO.listGroups(userInfo) map { groupMemberships =>
+      groupMemberships map { groupMembership =>
+        groupMembership.groupName
+      } intersect FireCloudConfig.ElasticSearch.discoverGroupNames
+    }
```

Comment:
you could make this a bit shorter - your call on code style if you want to implement this way or not:
```
    samDAO.listGroups(userInfo) map { groupMemberships =>
      groupMemberships map (_.groupName) intersect FireCloudConfig.ElasticSearch.discoverGroupNames
    }
```

Reply:
i tried several different ways to simplify - i knew you would have the winning one!